### DISCUSSION TITLE
Stuck on contract deployment /w injected wallets

### DISCUSSION THREAD

**Discussion by JaniAnttonen** [seanjameshan/starknet.js#511] [Q&A]
How to deploy (any) contracts with user-provided injected wallets? Should I construct transactions (declare & deploy) that get signed by the user's wallet but gets broadcasted with our own configuration that has all the needed methods? How does one do this? Trying to create an UI which enables users to deploy their own contracts (with predefined classes).

Already asked on Discord, did not get an answer, so dropping this here if someone knows.
Related issue: https://github.com/eqlabs/starknet-multisig-ui/issues/9

**Comment by tabaktoni** [comment]
Did you manage to resolve this ? If not please open the Issue.
